addons/binding/org.openhab.binding.regoheatpump/src/main/java/org/openhab/binding/regoheatpump/internal/RegoHeatPumpHandlerFactory.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.regoheatpump.internal;
+
+import static org.openhab.binding.regoheatpump.RegoHeatPumpBindingConstants.*;
+
+import java.util.Collections;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandlerFactory;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.openhab.binding.regoheatpump.handler.IpHusdataHandler;
+import org.openhab.binding.regoheatpump.handler.IpRego6xxHeatPumpHandler;
+import org.openhab.binding.regoheatpump.handler.SerialHusdataHandler;
+import org.openhab.binding.regoheatpump.handler.SerialRego6xxHeatPumpHandler;
+
+/**
+ * The {@link RegoHeatPumpHandlerFactory} is responsible for creating things and thing
+ * handlers.
+ *
+ * @author Boris Krivonog - Initial contribution
+ */
+public class RegoHeatPumpHandlerFactory extends BaseThingHandlerFactory {
+
+    private final static Set<ThingTypeUID> SUPPORTED_THING_TYPES_UIDS = Collections.unmodifiableSet(Stream
```

Comment:
Non-preferred order of modifiers (It was like this in a template, which has been changed recently) correct order is `public static final` See also: http://stackoverflow.com/questions/16731240/what-is-a-reasonable-order-of-java-modifiers-abstract-final-public-static-e#16732059